### ISSUE TITLE
Send DM to user when post is modded

### ISSUE THREAD

**Issue by chris-y** [fnic-lemmy/modreportbot#9]
I've just seen that another auto mod bot does this. Looks quite useful if we can notify the user that something has been removed and the reason.

**Comment by M4rtineski** [comment]
Yeah, that's one of my problems with lemmy that it doesn't happen. Which automod does that alteady?

**Comment by chris-y** [comment]
It's an auto mod on lemmy.world

**Comment by M4rtineski** [comment]
Wait, they have an instancewide automod doing that? That's cool!

**Comment by M4rtineski** [comment]
If you were to go with the idea I think it would be good to implement exceptions setting so we don't end up in a situation where user gets multiple messages for the same thing from different bot/mod accounts.

**Comment by chris-y** [comment]
Yes, lemmy.world users in particular will need to be filtered out of notifications.

I've had a look at Pythorhead and the API.  I can't see any modlog stuff in Pythorhead, and in the API under the PostView it doesn't look like moderation reasons are included there - but I'll need to look at some live data to confirm that as I thought I'd seen it there.

**Comment by chris-y** [comment]
Might be related to #6

**Comment by M4rtineski** [comment]
Maybe contacting db0 to implement this instance-wide would be worth a shot? Is LW's bot open source?

E: Oh wait, I got a little confused there, this wouldn't cover the communities outside of db0.

**Comment by chris-y** [comment]
I did a little bit of testing.  If I remove a post it removes it from all the list views through the API.  If I get the post directly using the ID it shows as removed:True but no reason in the response.  Looks like the only way to get the reason is through the modlog.

**Comment by M4rtineski** [comment]
That... Sounds like a good feature request for lemmy side to be done. Do you think if it's worth requesting?

**Comment by chris-y** [comment]
> That... Sounds like a good feature request for lemmy side to be done. Do you think if it's worth requesting?

Ideally the option to notify users when posts have been modded (and users banned) would be most useful to request.

**Comment by chris-y** [comment]
For my reference, this is an example ModRemovePost modlog:

```
{
    "removed_posts": [
        {
            "mod_remove_post": {
                "id": 22654,
                "mod_person_id": 1243606,
                "post_id": 13333615,
                "reason": "testing",
                "removed": true,
                "when_": "2025-01-23T11:34:09.420500Z"
            },
            "post": {
                "id": 13333615,
                "name": "Testing",
                "body": "test",
                "creator_id": 116594,
                "community_id": 387954,
                "removed": true,
                "locked": false,
                "published": "2024-06-14T13:06:11.609489Z",
                "deleted": false,
                "nsfw": false,
                "ap_id": "https://feddit.uk/post/13333615",
                "local": true,
                "language_id": 0,
                "featured_community": false,
                "featured_local": false
            },
            "community": {
                "id": 387954,
                "name": "bot_test",
                "title": "Bot testing",
                "removed": false,
                "published": "2024-06-14T13:04:13.786217Z",
                "deleted": false,
                "nsfw": false,
                "actor_id": "https://feddit.uk/c/bot_test",
                "local": true,
                "hidden": false,
                "posting_restricted_to_mods": false,
                "instance_id": 1,
                "visibility": "Public"
            }
        },
```

**Comment by M4rtineski** [comment]
> > That... Sounds like a good feature request for lemmy side to be done. Do you think if it's worth requesting?
> 
> Ideally the option to notify users when posts have been modded (and users banned) would be most useful to request.

Yeah but that request was already done (I think?) and visiting a link from an old thread to find out the post is gone would be the use scenario. It would just be more user friendly to return the modlog reason.

**Comment by chris-y** [comment]
> > > That... Sounds like a good feature request for lemmy side to be done. Do you think if it's worth requesting?
> > 
> > 
> > Ideally the option to notify users when posts have been modded (and users banned) would be most useful to request.
> 
> Yeah but that request was already done (I think?) and visiting a link from an old thread to find out the post is gone would be the use scenario. It would just be more user friendly to return the modlog reason.

With my dev hat on, I would say that modlog_id which references the modlog item which caused the removal would be the thing they need to add.  The actual reason can be looked up from there.  The Lemmy UI would need to be updated to look at this and display the reason.

Of course you could just look up the modlog entries for a particular post - that's already possible - so this info could probably be added to the UI anyway, when visiting a removed post.
